inc/uninstall.class.php

```diff
@@ -469,6 +473,70 @@ static function deleteFusionInventoryLink($itemtype, $items_id) {
       }
    }
 
+   /**
+    * Function to remove GLPI Inventory information for an asset
+    *
+    * @param CommonDBTM $item
+    *
+    * @return nothing
+   **/
+   static function deleteGlpiInventoryLink($item) {
+
+      $plug = new Plugin();
+      if ($plug->isActivated('glpiinventory') && function_exists('plugin_pre_item_purge_glpiinventory')) {
+
+         // let glpi-inventory to clean item if needed (agent / collect etc ..)
+         plugin_pre_item_purge_glpiinventory($item);
+      } else {
+         $agent = new Agent();
+         $agent->deleteByCriteria([
+            'itemtype' => $item->getType(),
+            'items_id' => $item->getID(),
+         ]);
+      }
+
+      // purge lock manually because related computer is not purged
+      $lockedfield = new Lockedfield();
+      if ($lockedfield->isHandled($item)) {
+         $lockedfield->itemDeleted();
+      }
+
+      // unlock item relations
+      global $DB;
+      $RELATION = getDbRelations();
+      if (isset($RELATION[$item->getTable()])) {
+         foreach ($RELATION[$item->getTable()] as $tablename => $field) {
```

Comment:
You should verify if related items have a `is_dynamic` property (can be done using `CommonDBTM::canBeDynamic()`.